### PR TITLE
feat: send telemetry when user installs or removes extension

### LINKED ISSUE #3018: Send telemetry when extension installed, removed and installation/removal errors
### Is your enhancement related to a problem? Please describe

Telemetry should collect information about installing and removing extensions. Installation/removal errors should be also reported. Implementation should be aware of automatically installed updates and does not count them as installations.

Current implementation only track:
- updates availability
- automatic and manual updates for installed extensions

### Describe the solution you'd like

It would be good to track only user triggered installs/removals so we can evaluate extension popularity based on developers' usage. The simplest way to avoid multiple changes explained in option 1 is place telemetry calls on UI side using TelemetryService class and window.telemetryTrack method. This kind of implementation would not require to recognize extension removal and install calls executed from extension updater automatically.

### Describe alternatives you've considered

Option 1:

[Extension Installer ](https://vscode.dev/github/dgolovin/podman-desktop/blob/main/packages/main/src/plugin/index.ts#L1653)should have reference to Telemetry instance and probably additional boolean parameter for installFromImage method to mark installations triggered by automatic update. That would let to avoid sending extension install telemetry for automatic and manual updates.

Option 2:

Implement automatic telemetry reporting for podman-desktop commands. 
Implement Install and remove extension commands.
Migrate UI to use commands instead of calling install/remove extension methods directly.

Simple automatic reporting would send telemetry about command execution that include:
- command id
- result of execution (success/failure)
- error in case of failure

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dgolovin**
### Screenshot/screencast of this PR

This PR add telemetry for user requests to install or remove extension. 

### What issues does this PR fix or reference?

Fix #3018.

### How to test this PR?

Enable telemetry, install and remove extension, then check telemetry reporting for new events 'extension-install' and 'extension-remove'.

### CODE REVIEW

File: packages/renderer/src/TelemetryService.ts [hunk 1]
```diff
@@ -46,4 +51,8 @@ export class TelemetryService {
       }
     }, 200);
   }
+
+  public handleExtensionEvent(event: ExtensionTelemetryEventType, id: string, featured: boolean, error: string) {
+    window.telemetryTrack(event, { id, featured, error });
+  }
```
Comment (benoitf) [inline packages/renderer/src/TelemetryService.ts:57]: IMHO helper for extension shouldn't be in the service, helper is where the type is known

Comment (jeffmaury) [reply packages/renderer/src/TelemetryService.ts:57]: Telemetry is handled backend in a general manner

**Comment by benoitf** [comment]
@dgolovin are you gonna handle the remarks by @jeffmaury ?

**Comment by dgolovin** [comment]
@benoitf yes, I have to implement telemetry for install/remove ether on main side, because UI not always has all info about extension.

**Comment by benoitf** [comment]
@dgolovin what is the state of this PR ?

**Comment by dgolovin** [comment]
Replacing it with two separate PR's:
* extension removal https://github.com/containers/podman-desktop/pull/3541
* extension install is coming later

**Pull request closed without merging**